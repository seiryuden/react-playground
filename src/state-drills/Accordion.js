import React from "react";

class Accordion extends React.Component{

    static defaultProps = {sections: []}

    constructor(props){

        super(props)

        console.log(this.props)

        this.state = {
            currentSectionIndex: null,
            buttonsClicked: 0
        }

        console.log(this.state)

    }

   
    renderSections(){
        console.log("renderSections ran")

        console.log(this.props.sections)

        let sections = this.props.sections.map((section, index) =>(
            
            <li key={index}>
                {!this.props.length && <button onClick={() => this.handleClick(index)}>
                    {section.title}
                </button>}
                {<p>{this.renderContent(index)}</p>}
                
            </li>
            
            )

        )
            
        return sections

    }


    handleClick = (index) =>{

        console.log("handleClick ran");
        this.setState({
            currentSectionIndex: index,
            
        }
            );
        
        console.log("handleClink end")
    }


    renderContent(index){

        console.log("renderContent ran");
        console.log(index);
        if(this.state.currentSectionIndex === index){
            
            return this.props.sections[index].content;

        } else { return null}

    }

    

    render(){
        
        return(
    
            <ul>
                {this.renderSections()}
            </ul>
    
        )
    
    }

}

export default Accordion;