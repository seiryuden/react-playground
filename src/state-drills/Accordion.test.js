import React from "react";
import ReactDOM from "react-dom";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Accordion from "./Accordion";





describe ("Accordion Component", () => {


    const sections = [
        {
          title: 'Section 1',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
          title: 'Section 2',
          content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
          title: 'Section 3',
          content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
      ]


    it("renders without crashing", ()=>{

        const div = document.createElement("div");
        ReactDOM.render(<Accordion />, div);
        ReactDOM.unmountComponentAtNode(div);

    })


    // INCOMPLETE!!  -  Required by assignment  


    it("renders empty <li> when given no props", ()=>{

        const wrapper = shallow(<Accordion />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it("renders no sections active by default", ()=>{

        const wrapper = sha

    })



})

