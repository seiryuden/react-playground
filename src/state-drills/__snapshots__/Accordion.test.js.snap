// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component renders empty <li> when given no props 1`] = `<ul />`;
